tests/files/tests.py

```diff
@@ -38,6 +40,23 @@ def test_context_manager(self):
         self.assertTrue(f.closed)
         self.assertTrue(orig_file.closed)
 
+    def test_open_resets_opened_file_to_start_and_returns_context_manager(self):
+        file = File(BytesIO(b'content'))
```

Comment:
I'm curious to learn what benefit testing with both `BytesIO` and `NamedTemporaryFile` offers?